### PR TITLE
Add FlatButton

### PR DESCRIPTION
**Pull request by curtinmartis**
* Add `widget` module
* Add `FlatButton` class for displaying a flat (or "borderless" as Android likes to call it) button that respects `colorButtonNormal` and `selectedItemBackground` from your app's theme
* Update build tools, targetSdk, and compileSdk
* Bump servos version

### CODE REVIEW

File: widget/src/main/java/com/ticketmaster/servos/widget/FlatButton.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.ticketmaster.servos.widget;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.drawable.Drawable;
+import android.graphics.drawable.LayerDrawable;
+import android.support.v7.widget.AppCompatButton;
+import android.util.AttributeSet;
+
+public class FlatButton extends AppCompatButton {
```
Comment (bsberry) [inline widget/src/main/java/com/ticketmaster/servos/widget/FlatButton.java]: Could you Javadoc the class with just a brief explanation of why it exists and how to use it's special features?

Comment (curtinmartis) [inline widget/src/main/java/com/ticketmaster/servos/widget/FlatButton.java]: Yep, probably a good idea =)

**Comment by sheaam30** [comment]
+1

**Comment by patjackson52** [comment]
Seems really helpful given the trouble you went though.  Would be nice to have a readme with screenshot of widgets that is updated as widgets are added.
+1

**Comment by curtinmartis** [comment]
Addressed comments

**Comment by bsberry** [comment]
Looks good. Thanks for adding the docs. If anyone else is curious what the readme will look like when rendered: https://github.com/Ticketmaster/servos/tree/cm-flatButton/widget

Restarted the failed Travis build which seemed to be exogenous due to failing to fetch some libraries. Once that's passing, +1. :+1:

**Comment by patjackson52** [comment]
+1

**Pull request merged**